gateway-ha/src/main/java/io/trino/gateway/ha/config/HaGatewayConfiguration.java

```diff
@@ -225,4 +231,36 @@ public void setManagedApps(List<String> managedApps)
     {
         this.managedApps = managedApps;
     }
+
+    public List<String> getStatementPaths()
+    {
+        return statementPaths;
+    }
+
+    public void setAdditionalStatementPaths(List<String> statementPaths)
+    {
+        // remove trailing slashes to ensure predictable behavior when splitting on "/"
+        this.statementPaths = Streams.concat(ImmutableList.of(V1_STATEMENT_PATH).stream(),
+                statementPaths.stream().peek(s -> validateStatementPath(s, statementPaths)).map(s -> s.replaceAll("/+$", ""))).toList();
+    }
+
+    private void validateStatementPath(String statementPath, List<String> statementPaths)
+    {
+        if (statementPath.startsWith(V1_STATEMENT_PATH) ||
+                statementPaths.stream().filter(s -> !s.equals(statementPath)).anyMatch(s -> s.startsWith(statementPath))) {
+            throw new HaGatewayConfigurationException("Statement paths cannot be prefixes of other statement paths");
+        }
+        if (!statementPath.startsWith("/")) {
+            throw new HaGatewayConfigurationException("Statement paths must be absolute");
+        }
```

Comment:
I don't understand why a dedicated exception class was needed. Can we use `checkArgument` method instead?